src/service/CreditCardService.java

```diff
@@ -0,0 +1,83 @@
+package service;
+
+import model.CreditCard;
+
+import java.sql.*;
+import java.util.ArrayList;
+
+/**
+ * Created by janicelee on 2018-03-24.
+ */
+public class CreditCardService {
+    private Connection connection;
+
+    public CreditCardService(Connection connection) {
+        this.connection = connection;
+    }
+
+    public ArrayList<CreditCard> getCustomerCreditCards(String username) {
+        ArrayList<CreditCard> creditCards = new ArrayList();
+
+        try {
+            PreparedStatement pstmt = connection.prepareStatement(
+                "SELECT Credit_Card.* FROM Pays_With INNER JOIN Credit_Card ON Credit_Card.c_id=Pays_With.c_id  " +
+                        "WHERE username = ?"
+            );
+            pstmt.setString(1, username);
+            ResultSet rs = pstmt.executeQuery();
+            while(rs.next()) {
+                int cid = rs.getInt("c_id");
+                Date expDate = rs.getDate("exp_date");
+                String token = rs.getString("token");
+                String type = rs.getString("type");
+                int lastDigits = rs.getInt("last_digits");
+                creditCards.add(new CreditCard(cid, expDate, token, type, lastDigits));
+            }
+
+            pstmt.close();
+        } catch(SQLException ex) {
+            System.out.println("Message: " + ex.getMessage());
+        }
+
+        return creditCards;
+    }
+
+    public void addCustomerCreditCard(String username, CreditCard newCreditCard) {
+        try {
+            PreparedStatement pstmt = connection.prepareStatement(
+                "INSERT INTO Credit_Card VALUES(?, ?, ?, ?, ?)"
```

Comment:
For the token, should we have a random string generator? I assume the customer isn't going to be filling in the token field?

Reply:
yes, that will be in the caller of this function who will generate the token and then construct the CreditCard object to pass to this function.